syne_tune/blackbox_repository/conversion_scripts/scripts/yahpo_import.py

```diff
@@ -251,6 +251,12 @@ def _objective_function(
             num_objectives = len(self.objectives_names)
             result = np.empty((num_fidelities, num_objectives))
             multiplier = 0.05 if self._is_iaml or self._is_rbv2 else 1
+            example_configuration = {
```

Comment:
Makes sense, thanks a lot for proposing this fix!

Given that the logic to get active hyperparameters can be a bit intricate, it may make sense to have a dedicated method `get_active_hyperparameters(configuration, fidelity)` that returns the list of active hyperparameters whether fidelity is None or not and which is called at the beginning of this `_objective_function` method, something like that:

```python
    def _objective_function(
        self,
        configuration: Dict[str, Any],
        fidelity: Optional[dict] = None,
        seed: Optional[int] = None,
    ) -> Dict[str, Any]:
        configuration = self._map_configuration(configuration.copy())
        active_hyperparameters = self._get_active_hyperparatemeters(configuration, fidelity)  # <- the new method
```

Let me know what you think.

Reply:
Yes, I also thought this could be done in a nicer way. @sighellan , are you up for this?

Reply:
Yes, I'll look at that!